vaadin-combo-box-flow-demo/src/main/java/com/vaadin/flow/component/combobox/demo/ComboBoxView.java

```diff
@@ -41,386 +51,314 @@
+        comboBox.setItemLabelGenerator(Information::getText);
         // end-source-example
 
-        comboBox.setId("lazy-loading-box");
-        addCard("Lazy Loading", "Lazy loading between client and server",
-                comboBox);
+        addCard("Presentation",
+                "Customizing drop down items with ComponentRenderer", comboBox);
     }
 
-    private void createComboBoxWithCallbackLazyLoading() {
+    private void usingTemplateRenderer() {
         //@formatter:off
         // begin-source-example
-        // source-example-heading: Lazy loading with callbacks
-        ComboBox<String> comboBox = new ComboBox<>();
+        // source-example-heading: Customizing drop down items with TemplateRenderer
+        ComboBox<Song> comboBox = new ComboBox<>();
+        comboBox.setLabel("Song");
+        List<Song> listOfSongs = createListOfSongs();
 
         /*
-         * This data provider doesn't load all the items to the server memory
-         * right away. The component calls the first provided callback to fetch
-         * items from the given range with the given filter. The second callback
-         * should provide the number of items that match the query.
+         * Providing a custom item filter allows filtering based on all of
+         * the rendered properties:
          */
-        comboBox.setDataProvider(
-                (filter, offset, limit) ->
-                    IntStream.range(offset, offset + limit)
-                        .mapToObj(i -> "Item " + i),
-                filter -> 500);
+        ItemFilter<Song> filter = (song, filterString) ->
+                song.getName().toLowerCase()
+                        .contains(filterString.toLowerCase())
+                        || song.getArtist().toLowerCase()
+                        .contains(filterString.toLowerCase());
 
+        comboBox.setItems(filter, listOfSongs);
+        comboBox.setClearButtonVisible(true);
+        comboBox.setItemLabelGenerator(Song::getName);
+        comboBox.setRenderer(TemplateRenderer.<Song>of(
+                "<div>[[item.song]]<br><small>[[item.artist]]</small></div>")
+                .withProperty("song", Song::getName)
+                .withProperty("artist", Song::getArtist));
         // end-source-example
         //@formatter:on
-        comboBox.setId("callback-box");
-        addCard("Lazy Loading", "Lazy loading with callbacks", comboBox);
-    }
 
-    private void createComboBoxWithCustomValues() {
-        Div message = createMessageDiv("custom-value-message");
+        comboBox.getStyle().set(ElementConstants.STYLE_WIDTH, WIDTH_STRING);
+        comboBox.setId("template-selection-box");
+        addCard("Presentation",
+                "Customizing drop down items with TemplateRenderer", comboBox);
+    }
 
+    private void styling() {
+
+        Div firstDiv = new Div();
+        firstDiv.setText(
+                "To read about styling you can read the related tutorial in");
```

Comment:
"To read about X you can read Y in" doesn't sound right.

Reply:
Also the text inside the info box doesn't wrap correctly: https://www.dropbox.com/s/v8uinmwliimd870/Screenshot%202019-08-22%2010.08.56.png?dl=0

...should probably look more like https://www.dropbox.com/s/24kwlk8msatcyc9/Screenshot%202019-08-22%2010.07.01.png?dl=0

Reply:
I pushed an update with the extra "in" word removed, capitalized "HTML" and fixed the markup so it wraps nicely.

The wording in both paragraphs here is still a bit weird and redundant imo but I think now it's kind of "ok". I would maybe prefer to just keep the links but remove the text in front of them since they don't add much value. Or write something completely different instead.